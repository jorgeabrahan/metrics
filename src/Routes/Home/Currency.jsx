import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const Currency = ({
  short, name,
}) => (
  <NavLink to={`/details/${short}`}>
    <div>
      <h2 className="currency__title">
        { name }
      </h2>
      <small>{ short }</small>
    </div>
    <div className="currency__icon">
      <span className="material-symbols-outlined">
        arrow_right_alt
      </span>
    </div>
  </NavLink>
);

Currency.propTypes = {
  short: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default Currency;
