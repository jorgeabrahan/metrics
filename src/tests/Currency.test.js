/* eslint-disable import/no-extraneous-dependencies */
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Currency from '../Routes/Home/Currency';
import wrap from './testResources';

describe('Tests for <Currency /> component', () => {
  const currency = (
    <Currency
      short="usd"
      name="United States Dollar"
    />
  );

  it('Should render the content correctly', () => {
    render(wrap(currency));
    expect(screen.getByText('usd')).not.toBeNull();
    expect(screen.getByText('United States Dollar')).not.toBeNull();
  });

  it('Should match the snapshot', () => {
    expect(renderer.create(wrap(currency)).toJSON()).toMatchSnapshot();
  });
});
