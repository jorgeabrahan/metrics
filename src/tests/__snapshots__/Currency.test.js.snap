// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Currency /> component Should match the snapshot 1`] = `
<a
  className=""
  href="/details/usd"
  onClick={[Function]}
>
  <div>
    <h2
      className="currency__title"
    >
      United States Dollar
    </h2>
    <small>
      usd
    </small>
  </div>
  <div
    className="currency__icon"
  >
    <span
      className="material-symbols-outlined"
    >
      arrow_right_alt
    </span>
  </div>
</a>
`;
